extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/structure/devices/deviceParameters/impl/DeviceBinaryInput.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.impl;
+
+import java.util.Map.Entry;
+
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.serverConnection.constants.JSONApiResponseKeysEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.deviceParameters.constants.DeviceBinarayInputEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.devices.impl.DeviceImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link DeviceBinaryInput} contains all information of an device binary input, e.g. binary input type id (see
+ * {@link DeviceBinarayInputEnum}, state and so on.
+ *
+ * @author Michael Ochel - initial contributer
+ * @author Matthias Siegele - initial contributer
+ *
+ */
+public class DeviceBinaryInput {
+
+    private Logger logger = LoggerFactory.getLogger(DeviceBinaryInput.class);
+
+    private Short targetGroupType = null;
+    private Short targetGroup = null;
+    private Short inputType = null;
+    private Short inputId = null;
+    private Short stateValue = null;
+
+    /**
+     * Creates a new {@link DeviceBinarayInputEnum} through the {@link JsonObject} of the binary inputs at json response
+     * from digitalSTROM JSON-API or property-tree. Will be automatically added to a {@link DeviceImpl}, if binary
+     * inputs exists.
+     *
+     * @param jsonObject must not be null
+     */
+    public DeviceBinaryInput(JsonObject jsonObject) {
+        try {
+            for (Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+                try {
+                    getClass().getDeclaredField(entry.getKey()).set(this, entry.getValue().getAsShort());
+                } catch (NoSuchFieldException e) {
+                    // ignore
+                } catch (NumberFormatException e) {
+                    // ignore
+                }
+            }
+            if (stateValue == null && jsonObject.get(JSONApiResponseKeysEnum.STATE.getKey()) != null) {
+                stateValue = jsonObject.get(JSONApiResponseKeysEnum.STATE.getKey()).getAsShort();
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("A IllegalArgumentException occurred: ", e);
```

Comment:
This will end up in the users logfile who does not know what this is... Either change the severity to debug or if you want to let the user know so he can report the error, please give a description what the exception mean in this context. This also applies for the other exceptions you are catching here.